scripts/globals/job_utils/rune_fencer.lua

```diff
@@ -50,8 +50,166 @@ local function enforceRuneCounts(target)
     end
 end
 
+--source https://www.bluegartr.com/threads/124844-Vivacious-Pulse-testing?p=6038534&viewfull=1#post6038534 and following posts
+local function getRuneHealAmount(effect, target)
+
+    if effect == xi.effect.TENEBRAE then -- Tenebrae primarily restores MP, but also uses the base divine skill/2 bonus but no raw stat bonus.
+        return 0
+    end
+
+    local runeStatMap = {
+                            [xi.effect.IGNIS]  = xi.mod.STR,
+                            [xi.effect.GELUS]  = xi.mod.DEX,
+                            [xi.effect.FLABRA] = xi.mod.VIT,
+                            [xi.effect.TELLUS] = xi.mod.AGI,
+                            [xi.effect.SULPOR] = xi.mod.INT,
+                            [xi.effect.UNDA]   = xi.mod.MND,
+                            [xi.effect.LUX]    = xi.mod.CHR,
+                        }
+
+    local stat = runeStatMap[effect:getType()]
+    if stat ~= nil then
+        return math.floor(target:getStat(stat) * 0.5)
+    end
+
+    return 0
+end
+
+
+
+--source https://www.bg-wiki.com/ffxi/Vivacious_Pulse
+local function calculateVivaciousPulseHealing(target)
+
+    local divineMagicSkillLevel = target:getSkillLevel(xi.skill.DIVINE_MAGIC)
+    local HPHealAmount = 10 + math.floor(divineMagicSkillLevel / 2 * (100 + target:getJobPointLevel(xi.jp.VIVACIOUS_PULSE_EFFECT)) / 100) -- Bonus of 1-20%  from Vivacious pulse job points.
```

Comment:
Just I random thought, feel free to ignore it.
I think divineMagicSkillLevel / 2 should be floored if its gonna get multiplied right after. SE usually floors after each step.
I have no proof of this though.

Reply:
I went and checked retail, and I got a second opinion confirming it, see the following post and the post after
https://www.ffxiah.com/forum/topic/37086/endeavoring-to-awaken-a-guide-to-rune-fencer/180/#3615355